### ISSUE TITLE
¿Why the scaling factor of the embedding layer by sqrt hidden size?

### ISSUE THREAD

**Issue by Nikotarou** [tensorflow/tensor2tensor#1718]
### Description
I found that the hyperparameter multiply_embedding_mode can be chosen to scale the embeddings after they are looked up. I also found that this make sense as the initialization of the embeddings is a normal `(0, sqrt(hidden_dim))` in line 479 of modalities.py:
` tf.get_variable(
   var_name, [shard_size, hidden_dim],
   initializer=tf.random_normal_initializer(0.0, hidden_dim ** -0.5)))`
But why initializing with low std and then scale instead of directly using a specific initialization std. Is there some paper explaining this choice or it just empirical? As far as I understand by doing it this way we now have a scaling factor of `sqrt(hidden_dim)` in the gradients of the embedding layer, making them bigger in magnitude.

In my case I must to say that that the scaling is needed, as if it not present the accuracy drops hardly (maybe because it just needs more epochs to train). But still don't get the reason behind this. Hope someone can help me.

**Comment by abnf** [comment]
Maybe this is related to the curse of dimensionality. sqrt(dim) seems to be a usual scaling factor to prevent issues.

**Comment by calessandri** [comment]
The transformer shares the weights of the embedding layer and the linear transformation before the output softmax. The initializer seems to be set to have the right scale for the output linear transformation, so the embeddings are multiplied by a the scaling factor.
